### PR TITLE
[treehopper] Fix dependence libusb

### PR DESCRIPTION
**Pull request by FrankXie05**
Fix https://github.com/microsoft/vcpkg/pull/35435#discussion_r1413497744
Note: Fix correct `libusb` usage.
Error:
```
/treehopper/src/1.11.3-f321b6be5f.clean/C++/API/inc/LibUsbConnection.h:9:10: fatal error: 'libusb-1.0/libusb.h' file not found
#include <libusb-1.0/libusb.h>
         ^~~~~~~~~~~~~~~~~~~~~
1 warning and 1 error generated.

```

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [ ] ~~SHA512s are updated for each updated download~~
- [x] The "supports" clause reflects platforms that may be fixed by this new version
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [ ] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/treehopper/fix-dependences-libusb.patch [hunk 1]
```diff
@@ -0,0 +1,15 @@
+diff --git a/C++/API/CMakeLists.txt b/C++/API/CMakeLists.txt
+index 8537305..612cf7d 100644
+--- a/C++/API/CMakeLists.txt
++++ b/C++/API/CMakeLists.txt
+@@ -37,7 +37,9 @@ if(APPLE)
+     find_library(IOKIT IOKit)
+     target_link_libraries(treehopper pthread ${CORE_FOUNDATION} ${IOKIT})
+ elseif(UNIX)
+-    target_link_libraries(treehopper usb-1.0 pthread)
++    find_package(libusb CONFIG REQUIRED)
```
Comment (dg0yt) [inline ports/treehopper/fix-dependences-libusb.patch:10]:
```suggestion
+    find_package(libusb REQUIRED)
```
There is no config, so don't hard-code it. (There is not even a find module. Just a vcpkg wrapper.)

File: ports/treehopper/fix-dependences-libusb.patch [hunk 2]
```diff
@@ -0,0 +1,15 @@
+diff --git a/C++/API/CMakeLists.txt b/C++/API/CMakeLists.txt
+index 8537305..612cf7d 100644
+--- a/C++/API/CMakeLists.txt
++++ b/C++/API/CMakeLists.txt
+@@ -37,7 +37,9 @@ if(APPLE)
+     find_library(IOKIT IOKit)
+     target_link_libraries(treehopper pthread ${CORE_FOUNDATION} ${IOKIT})
```
Comment (dg0yt) [inline ports/treehopper/fix-dependences-libusb.patch:7]: FTR IMO this should also be patched, to decouple cached artifacts from local SDK paths. Untested:
```suggestion
     target_link_libraries(treehopper pthread "-framework CoreFoundation" "-framework IOKit")
```

**Comment by dg0yt** [comment]
Use libusb pkgconfig, not vcpkg's cmake package hack.

**Comment by FrankXie05** [comment]
The PR https://github.com/microsoft/vcpkg/pull/35498 dose not fix the usage.

**Review by Cheney-W — DISMISSED** [review]

**Comment by BillyONeal** [comment]
> Use libusb pkgconfig, not vcpkg's cmake package hack.

I'm pretty sure in general that we currently prefer the unofficial cmake package because we don't have a great answer for what happens for pkgconfig on Windows, as downstream consumers of us don't usually have a copy of pkgconfig that their CMake is going to find.

That also said, that doesn't apply in this case because the libusb backend is never selected on Windows.

**Comment by Neumann-A** [comment]
> because we don't have a great answer for what happens for pkgconfig on Windows, as downstream consumers of us don't usually have a copy of pkgconfig that their CMake is going to find.

Just depend on port `pkgconf` on windows and cmake will find that without problems.

**Comment by BillyONeal** [comment]
> Just depend on port `pkgconf` on windows and cmake will find that without problems.

Not in downstream's CMake

File: ports/treehopper/fix-dependences.patch [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
++++ b/C++/API/CMakeLists.txt
+@@ -37,7 +37,10 @@ if(APPLE)
+     find_library(IOKIT IOKit)
+     target_link_libraries(treehopper pthread ${CORE_FOUNDATION} ${IOKIT})
+ elseif(UNIX)
+-    target_link_libraries(treehopper usb-1.0 pthread)
++    find_package(PkgConfig REQUIRED)
++    pkg_check_modules(libusb REQUIRED IMPORTED_TARGET libusb-1.0)
++
++    target_link_libraries(treehopper PRIVATE PkgConfig::libusb pthread)
```
Comment (BillyONeal) [inline ports/treehopper/fix-dependences.patch:13]: I double checked that this is not named in the config because it is PRIVATE

**Review by BillyONeal — DISMISSED** [review]

**Comment by Neumann-A** [comment]
> Not in downstream's CMake

It does. Did you forgot about:
https://github.com/microsoft/vcpkg/blob/c6f9a7b205b7c6847513a60a9759e1e600e0646d/scripts/buildsystems/vcpkg.cmake#L557-L590

Since `CMAKE_PROGRAM_PATH` is modified `find_program` will actually find `pkgconf` if it is installed by vcpkg

**Comment by BillyONeal** [comment]
LLVM failure was a random ICE we aren't normally seeing in CI, so let's hope simply rerunning fixes it

**Comment by dg0yt** [comment]
> Just depend on port `pkgconf` on windows and cmake will find that without problems.

"Without problems" is not true.
Our cmake maintainer functions do not set `VCPKG_HOST_TRIPLET`. So the host pkgconf is not found without extra options.
Related discussion is (at least) in https://github.com/microsoft/vcpkg/pull/25529.

File: ports/treehopper/fix-dependences.patch (hunk 3 above)
Comment (dg0yt) [reply ports/treehopper/fix-dependences.patch:13]: It should be appear for static triplets as `$<LINK_ONLY:...>`.

**Pull request merged**